src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,462 @@
+            const candidate: MinimalUser = await addCandidate(partId, false);
+
+            // When the candidate removes themself from the list
+            const result: Promise<void> = joinerDAO.removeCandidate(partId, candidate);
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid changing its own candidate fields', async() => {
+            // Given a part and joiner, with a candidate
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId);
+
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious users tries to change one of its own candidate's fields
+            const update: Partial<MinimalUser> = { name: 'foo' };
+            const result: Promise<void> = joinerDAO.subCollectionDAO(partId, 'candidates').set(candidate.id, update);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid to change fields', async() => {
+            // Given a part, and a candidate (that is not the creator)
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId, false);
+
+            const updates: Partial<Joiner>[] = [
+                { chosenOpponent: candidate },
+                { partStatus: PartStatus.CONFIG_PROPOSED.value },
+                { firstPlayer: FirstPlayer.CHOSEN_PLAYER.value },
+                { partType: PartType.BLITZ.value },
+                { maximalMoveDuration: 73 },
+                { totalPartDuration: 1001001 },
+            ];
+            for (const update of updates) {
+                // When modifying a field
+                const result: Promise<void> = joinerDAO.update(partId, update);
+
+                // Then it should fail
+                await expectPermissionToBeDenied(result);
+            }
+        });
+    });
+    describe('for chosen opponent', () => {
+        let partId: string;
+        beforeEach(async() => {
+            // Given a joiner where the current user is set as chosen opponent
+
+            // We have to follow a realistic workflow to achieve that
+            // The part is first created
+            const createdPartId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            partId = createdPartId;
+
+            // A candidate adds themself to the candidates list
+            const candidate: MinimalUser = await addCandidate(partId);
+
+            // The creator then selects candidates as the chosen opponent and proposes the config
+            await reconnectUser(CREATOR_EMAIL);
+            const update: Partial<Joiner> = { chosenOpponent: candidate, partStatus: PartStatus.CONFIG_PROPOSED.value };
```

Comment:
on sais pas faire ça pour l'instant, choisir ET proposer en même temps ! c'pourrais être bien ceci dit.

Reply:
hmm faire deux clics en un ? Niveau UX je suis pas sûr que ce soit intéressant (avoir un bouton "pick" et un bouton "pick and propose").
Niveau sécu par contre, c'est pas un soucis de l'autoriser même si c'est pas possible dans l'interface je dirais.

Reply:
ton avis UX est sûrement valide, bref, optimisation UX future V4 sûrement :p !

Reply:
ou non-optimisation plutôt :D On résouds ça sans ticketter, ou avec un ticket « demander à quelqu'un qui s'y connait en UX de donner son avis sur le site complet » ?

Reply:
ah bah ce serait clairement le must d'avoir une UX Person pour le site, c'est sûûûûr ! Tu peux donc ticketter !